### PR TITLE
Update python2 brew recipe

### PR DESCRIPTION
**Pull request by porttack**
* Upgrade to 2.7.17_1
* Tweak to work with current brew lint guidelines
* Remove unneeded tcl-tk switch (that breaks things)

Issue: https://khanacademy.atlassian.net/browse/INFRA-5871

Test plan:

`brew install --force-bottle python@2` (test in catalina before python is installed)

### CODE REVIEW

File: Formula/git-workflow.rb [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
 
   def install
-
     # git-find-reviewers specifically requires python2
     inreplace "bin/git-find-reviewers" do |s|
       s.gsub! "#!/usr/bin/env python", "#!/usr/bin/env python2"
     end
 
-    %W[
+    %w[
```
Comment (dat-boris) [inline Formula/git-workflow.rb:15]: Ha - I'm no ruby expert - but this makes sense from searching.

**Review by dat-boris — APPROVED** [review]
LGTM! I would love @dbraley input also, as I'm no ruby or brew expert 😅

File: Formula/git-workflow.rb (hunk 1 above)
Comment (porttack) [reply Formula/git-workflow.rb:15]: Linting refused to accept the `%W`. I searched around and couldn't figure it out either. If we have a regression, I'll figure it out.

File: Formula/arcanist.rb [hunk 2]
```diff
@@ -35,9 +37,5 @@ def install
 
     # move khan-bin into bin
     mv prefix/"khan-bin/arc", bin/"arc"
-
-  end
-
-  test do
```
Comment (dbraley) [inline Formula/arcanist.rb:41]: Aw, you removed this test? It seems so useful! 🤣 🙅🏻

File: Formula/git-workflow.rb (hunk 1 above)
Comment (dbraley) [reply Formula/git-workflow.rb:15]: Unless i'm mistaken, the `W` allows for variable expansion, `w` does not. You don't have any variables in the array, so no need to allow for expansion.

**Review by dbraley — APPROVED** [review]
This seems good by me. I'll admit to not really knowing how to configure brew formula, but a quick google indicates this is correct! Or at least as correct as I'm going to be able to grok right now. Do we need to use a cask here to brew install this? Or is that already set up in dotfiles?

**Pull request closed without merging**